haystack/components/generators/chat/openai.py

```diff
@@ -511,6 +572,10 @@ def _convert_chat_completion_to_chat_message(completion: ChatCompletion, choice:
             "usage": _serialize_usage(completion.usage),
         },
     )
+    # Using pdantic with structured output, openai also returns the message.parsed
+    # return the parsed message in the meta
+    if isinstance(message, ParsedChatCompletionMessage) and message.parsed:
+        chat_message.meta["parsed"] = message.parsed
```

Comment:
In my opinion, we can fix #9776 by simply avoiding storing this information.

1. My impression is that most users need the parsed/extracted information, so a JSON string seems OK
2. As we investigated earlier in this PR, serializing Pydantic objects can be done easily but deserialization is problematic (unless we save/serialize the original `BaseModel`, which would make things much more complex)
3. (If there are many requests to introduce this aspect, we can rethink about this decision in a non-breaking way)